docs/onboarding/Onboarding.md

```diff
@@ -0,0 +1,125 @@
+## Onboarding guide
+
+Playbook Dispatcher is a service that simplifies running Ansible Playbooks on connected hosts.
+This guide explains how an application be integrated with it in order to benefit from:
+
+* common API for dispatching playbooks
+* automated tracking of playbook state
+* common model for representing playbook results
+* built-in authorization support
+
+### Dispatching playbooks
+
+Currently, dispatching of playbooks is only possible with the [Internal/Private API](https://github.com/RedHatInsights/playbook-dispatcher/blob/master/schema/private.openapi.yaml), and it will only be available to accept requests within the cluster.
+
+#### Authenticating with the Internal API
+
+In order to send requests to the Internal API, the client is required to provide a service specific pre-shared key under the `Authorization` header of the HTTP request.
+
+For example:
+```javascript
+POST /internal/dispatch HTTP/1.1
+Authorization: PSK <pre-shared key>
+```
+
+The service specific pre-kshared keys are configured via environment variables.
+You can open a PR in this repository to add a dispatcher pre-shared key for your service.
+The environment variable will need to be in the form of `PSK_AUTH_<service id>`, for example, `PSK_AUTH_REMEDIATIONS`.
+Here is a PR that added a PSK for the Tasks app [#217](https://github.com/RedHatInsights/playbook-dispatcher/pull/217).
+
+#### Obtaining the necessary infomation
+
+To dispatch a playbook, you will need to POST to the private api's `/internal/v2/dispatch` endpoint.
+The openapi schema for this endpoint can be found [here](https://github.com/RedHatInsights/playbook-dispatcher/blob/master/schema/private.openapi.yaml#L77).
+
+The endpoint accepts the following parameters:
+* `recipient`
+  * An uuid used by [Cloud Connector](https://github.com/RedHatInsights/cloud-connector) to route the playbook signal. This field is required.
+    * For `rhc-worker-playbook` or "directly connected hosts", `recipient` can be obtained from the [system profile](https://github.com/RedHatInsights/inventory-schemas/blob/8000191d960da05c4ebf7960f4af8f7cf68bf616/schemas/system_profile/v1.yaml#L197).
+    * For satellite hosts, this will be provided by the [Sources Api](https://github.com/RedHatInsights/sources-api-go), where it is know as the `rhc_id`.
+    The `rhc_id`, or `recipient`, can be obtained using the `/sources/{id}/rhc_connections` [endpoint](https://github.com/RedHatInsights/sources-api-go/blob/9a7c9288be53f84717a6337063a481dcf533f615/public/openapi-3-v3.1.json#L1333).
```

Comment:
What is the `{id}` in `/sources/{id}/rhc_connections`? Having a host in inventory, where do I get this id from?